common/persistence/serialization/thrift_mapper_test.go

```diff
@@ -0,0 +1,482 @@
+		DecisionRequestID:                  common.StringPtr("DecisionRequestID"),
+		CancelRequestID:                    common.StringPtr("CancelRequestID"),
+		StickyTaskList:                     common.StringPtr("StickyTaskList"),
+		StickyScheduleToStartTimeout:       common.DurationPtr(time.Minute * time.Duration(rand.Intn(10))),
+		RetryAttempt:                       common.Int64Ptr(int64(rand.Intn(1000))),
+		RetryInitialInterval:               common.DurationPtr(time.Minute * time.Duration(rand.Intn(10))),
+		RetryMaximumInterval:               common.DurationPtr(time.Minute * time.Duration(rand.Intn(10))),
+		RetryMaximumAttempts:               common.Int32Ptr(int32(rand.Intn(1000))),
+		RetryExpiration:                    common.DurationPtr(time.Minute * time.Duration(rand.Intn(10))),
+		RetryBackoffCoefficient:            common.Float64Ptr(rand.Float64() * 1000),
+		RetryExpirationTimestamp:           common.TimePtr(time.Now()),
+		RetryNonRetryableErrors:            []string{"RetryNonRetryableErrors"},
+		HasRetryPolicy:                     common.BoolPtr(true),
+		CronSchedule:                       common.StringPtr("CronSchedule"),
+		EventStoreVersion:                  common.Int32Ptr(int32(rand.Intn(1000))),
+		EventBranchToken:                   []byte("EventBranchToken"),
+		SignalCount:                        common.Int64Ptr(int64(rand.Intn(1000))),
+		HistorySize:                        common.Int64Ptr(int64(rand.Intn(1000))),
+		ClientLibraryVersion:               common.StringPtr("ClientLibraryVersion"),
+		ClientFeatureVersion:               common.StringPtr("ClientFeatureVersion"),
+		ClientImpl:                         common.StringPtr("ClientImpl"),
+		AutoResetPoints:                    []byte("AutoResetPoints"),
+		AutoResetPointsEncoding:            common.StringPtr("AutoResetPointsEncoding"),
+		SearchAttributes:                   map[string][]byte{"key_1": []byte("SearchAttributes")},
+		Memo:                               map[string][]byte{"key_1": []byte("Memo")},
+		VersionHistories:                   []byte("VersionHistories"),
+		VersionHistoriesEncoding:           common.StringPtr("VersionHistoriesEncoding"),
+	}
+	actual := workflowExecutionInfoFromThrift(workflowExecutionInfoToThrift(expected))
+	assert.Equal(t, expected.ParentDomainID, actual.ParentDomainID)
+	assert.Equal(t, expected.ParentWorkflowID, actual.ParentWorkflowID)
+	assert.Equal(t, expected.ParentRunID, actual.ParentRunID)
+	assert.Equal(t, expected.InitiatedID, actual.InitiatedID)
+	assert.Equal(t, expected.CompletionEventBatchID, actual.CompletionEventBatchID)
+	assert.Equal(t, expected.CompletionEvent, actual.CompletionEvent)
+	assert.Equal(t, expected.CompletionEventEncoding, actual.CompletionEventEncoding)
+	assert.Equal(t, expected.TaskList, actual.TaskList)
+	assert.Equal(t, expected.WorkflowTypeName, actual.WorkflowTypeName)
+	assert.True(t, (*expected.WorkflowTimeout-*actual.WorkflowTimeout) < time.Second)
```

Comment:
Can you explain why this is needed? Do we loose precision with rounding on conversion?

Reply:
Yes we only store at seconds. But in practice we won't actually lose anything because we will also truncate to seconds at the RPC layer.